main.py

```diff
@@ -111,26 +111,21 @@ def produce_kafka_messages(bucket, job_id, fixture_data, args):
     )
     s3_client = boto3.client("s3")
     for s3_key in fixture_data:
-        logger.debug(f"Processing key: {s3_key}")
-        line_no = 0
-        for line in (
-            s3_client.get_object(Bucket=bucket, Key=s3_key)["Body"]
-            .read()
-            .splitlines()
-        ):
-            line_no += 1
-            try:
-                data = json.loads(line)
-                db = data["message"]["db"] if "db" in data["message"] else "missingDb"
-                collection = data["message"]["collection"] if "collection" in data["message"] else "missingCollection"
-                topic_name = f"{args.topic_prefix}{job_id}_{db}.{collection}"
-            except json.JSONDecodeError as err:
-                logger.error(
-                    f"line {line_no} of {s3_key} contains invalid JSON data: {err.msg}"
-                )
-                continue
-            producer.send(topic_name, line)
-            producer.flush()
+        logger.info(f"Processing key: {s3_key}")
+        file = s3_client.get_object(Bucket=bucket, Key=s3_key)["Body"].read()
+        try:
+            data = json.loads(file)
+            db = data["message"]["db"] if "db" in data["message"] else "missingDb"
+            collection = data["message"]["collection"] if "collection" in data["message"] else "missingCollection"
+            topic_name = f"{args.topic_prefix}{job_id}_{db}.{collection}"
+        except json.JSONDecodeError as err:
```

Comment:
I think this should just be a generic exception handler, just in case something really weird goes on trying to parse the file.  I *think* it's all handled by the various `if`s in there but who knows what destructive testing someone might do!  Certainly trying to catch `json.JSONDecodeError` isn't right as we're no longer doing any JSON parsing.  Might want to remove the json import at the top too now.

Reply:
will do